_test/test_data_loaders/test_nxspepar_loader.m

```diff
@@ -96,6 +97,21 @@ function test_mslice_par(this)
             [~,fname,fext] = fileparts(al.par_file_name);
             assertEqual([fname,fext],'MAP11014v2.nxspe');
         end
+        %
+        function test_mslice_par_cached(this)
```

Comment:
Would `test_you_can_reload_a_cached_mslice_nxspe_par_file` be a more descriptive test name here?

Reply:
`build_sqw_in_parallel` is completely unrelated to this. 
as for name -- your suggestion is confusing. 
I would rather name it `mslice_par_from_cache_correct`  (when data are loaded from cache, they are still returned in correct format -- this was the issue in the first place).  It close to what I wrote, but if sounds better, I am changing it.

Reply:
The first two commits of the PR come from some fixes done to release 3_4 done ages ago. We have not merged them into master directly, so I do not remember when it was done and what was discussed. Changes are simple and reasonable -- A user normally should not gen_sqw in parallel immediately -- I, for example, always do something wrong in couple of first runs.